### DISCUSSION TITLE
@next/third-parties/google is blocking instead of loading after everything else

### DISCUSSION THREAD

**Discussion by JPVRS** [vercel/next.js#66834] [Help]
### Summary

Using the pages router here. We're loading a GTAG using @next/third-parties/google by importing it in our _app component as instructed in the docs:

https://nextjs.org/docs/pages/building-your-application/optimizing/third-party-libraries

It works in terms of the tag being detected properly in Google Tag Manager. But it's blocking for a significant amount of time according to Page Speed Insights (hundreds of milliseconds). The docs say it's supposed to load after everything else is done.

Is there an issue with this library? Should we be doing something else instead? This is contributing to a performance issue for us.

Thanks in advance.

**Comment by JetAMC** [comment]
Same issue for me. I have done like it's described in the documentation, but TBT is huge:
![image](https://github.com/vercel/next.js/assets/42616753/c00e2fd4-7dd6-4953-87c3-cb4d2282d360)

**Comment by 100tomtomas100** [comment]
I have the same issue it delays page loading by 2sec and istead of being the last it loads first

**Comment by rosaldo260** [comment]
Yeah, I can't find solution how to load the tag manager without slowing down the site.

Any solutions?
